apps/06.1Checkers.js

```diff
@@ -3,75 +3,190 @@
-        // if the location is "truthy" (contains a checker piece, in this case)
-        if (this.grid[row][column]) {
-          // push the symbol of the check in that location into the array
-          rowOfCheckers.push(this.grid[row][column].symbol);
-        } else {
-          // just push in a blank space
-          rowOfCheckers.push(' ');
-        }
-      }
-      // join the rowOfCheckers array to a string, separated by a space
-      string += rowOfCheckers.join(' ');
-      // add a 'new line'
-      string += "\n";
-    }
+ // we start with our row number in our array
+    var rowOfCheckers = [row];
+// a loop within a loop
+    for (var column = 0; column < 8; column++) {
+// if the location is "truthy" (contains a checker piece, in this case)
+    if (this.grid[row][column]) {
+                    // push the symbol of the check in that location into the array
+    rowOfCheckers.push(this.grid[row][column].symbol);
+}   else {
+                    // just push in a blank space
+    rowOfCheckers.push(' ');
+}
+}
+            // join the rowOfCheckers array to a string, separated by a space
+    string += rowOfCheckers.join(' ');
+            // add a 'new line'
+    string += "\n";
+}
     console.log(string);
-  };
+};
 
-  // Your code here
+    //create the array thats going to hold the coordinates of checkers
+    this.checkers = [];
+    this.createCheckers = function () {
+            //create two arrays for initial positions of black and white checkers
+    var whitePositions = [];
+    var blackPositions = [];
+            //fill the two initial positions arrays with the coordinates of the checkers 
+    var j = 0;
+    while (j < 3) {
+    for (var i = 1; i <= 7; i = i + 2) {
+    whitePositions[j] = [0, i];
+    whitePositions[j + 8] = [2, i];
+    blackPositions[j + 4] = [6, i];
+    j++;
+}
+    j = 0;
+    for (var i = 0; i < 7; i = i + 2) {
+    whitePositions[j + 4] = [1, i];
+    blackPositions[j] = [5, i];
+    blackPositions[j + 8] = [7, i];
+    j++;
+}
+}
+            //create checkers and put the instances into the grid and into an array checkers
+    for (i = 0; i < 12; i++) {
+    this.checker = new Checker("white");
+    var x = whitePositions[i][0];
+    var y = whitePositions[i][1];
+    this.grid[x][y] = new Checker("white");
+    this.checkers.push(this.grid[x][y]);
+            }
+    for (i = 0; i < 12; i++) {
+    this.checker = new Checker("black");
+    var x = blackPositions[i][0];
+    var y = blackPositions[i][1];
+    this.grid[x][y] = new Checker("black");
+    this.checkers.push(this.grid[x][y]);
+}
 }
+        //helper function to return the instance in the grid at the correct row, column
+    this.selectChecker = function (row, column) {
+    return this.grid[row][column];
+        }
+        //this checks for a winner
+    this.checkForWinner = function () {
+        //using some, see if a value exists in the array
+    if (!this.checkers.some(function (val) {
```

Comment:
Nice use of the some() method but you should not call a return statement inside an `if` statement.

Format your logic like so: 

``` 
this.checkForWhiteCheckers = function(checkerPiece) {
      return checkerPiece.symbol === String.fromCharCode(0x1263B);
    }
// ...seperate logic for blackCheckers here ..

this.checkForWinner = function () {
      if (!this.checkers.some(this.checkForWhiteCheckers)){
          console.log("black wins!");
          process.exit();
      }
      if(!this.checkers.some(this.checkForBlackCheckers)) {
          console.log("white wins!");
      }
    }
```